### PR TITLE
Message syntax highlighting

### LINKED ISSUE #4677: Support code syntax highlighting in message code blocks
We support this in thread code blocks, is it possible to get consistent behavior in message code blocks now that we're plain-text across the board?

### PR DESCRIPTION
**Pull request by Ponjimon**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api
- hyperion (frontend)

**Release notes for users (delete if codebase-only change)**
- Messages now also support syntax highlighting for code blocks

**Related issues (delete if you don't know of any)**
Closes #4677

### CODE REVIEW

File: shared/draft-utils/process-message-content.js
Comment (mxstbr) [inline shared/draft-utils/process-message-content.js]: The `type` argument here should be lower case to match the input `message.messageType`. 

That way, you do not have to have a conditional outside of this method and can simply do

```JS
message.content.body = processMessageContent(message.messageType, message.content.body);
message.messageType = 'draftjs'
```

**Review by mxstbr — COMMENTED** [review]
This is super great, thank you!! :pray: 

I have one tiny code nitpick inline, but overall the extraction of the common methods makes total sense, perfectly done 👍 Once that is cleaned up we can ship this!

File: shared/draft-utils/process-message-content.js
Comment (Ponjimon) [reply shared/draft-utils/process-message-content.js]: To be honest, I just copied most of this file from here: https://github.com/withspectrum/spectrum/blob/alpha/shared/draft-utils/process-thread-content.js#L6 and removed one property that was irrelevant for messages. Could have done this refactor better probably, but I just didn't want to change the already existing naming of the stuff. If I change this here, it should be changed everywhere else too. 

The `type` argument / property for thread-related stuff is uppercase in the code. Is it lowercase like this anywhere else? Otherwise I'd suggest just uppercasing it here for the sake of consistency.

Comment (mxstbr) [reply shared/draft-utils/process-message-content.js]: It is inconsistent in the database, but it should be consistent with that. Threads have a `type` property that is either `TEXT` or `DRAFTJS`, but messages have a `messageType` property that is either `text` or `draftjs`.

That is inconsistent with each other, which is bad, but this method shiuld be consistent with the database, if that makes sense?

Comment (Ponjimon) [reply shared/draft-utils/process-message-content.js]: Oh, I was not aware of that, I guess. Will change it then.

Comment (mxstbr) [reply shared/draft-utils/process-message-content.js]: Yep I figured, that's why I left that review comment. Sorry for not giving the full context in the initial comment!

Comment (Ponjimon) [reply shared/draft-utils/process-message-content.js]: Okay, I committed some changes to fix it. While on it, I also noticed that those strings (`'text'` and `'draftjs'`) and so on, are all over the code, so I made some sort of "enum" for it and replaced all occurencies I could find using that. Strings are prone to typos, this should avoid it for future use. I can revert that and just fix the above, if that's required, but I think it's a good addition.

**Review by mxstbr — DISMISSED** [review]
This is great, thanks for making that more consistent! :100: I think we can ship this, will defer to @brianlovin for a final review

**Comment by brianlovin** [comment]
@lookapanda this is really good, but is it possible for us to preserve the language selection when editing a message? If I type a message with a js-highlighted code block, then edit it and save the edit, the js highlighting goes away.

**Comment by Ponjimon** [comment]
Let me check

**Comment by Ponjimon** [comment]
Okay, I figured out how it works. I needed to modify the `stateToMarkdown` function a bit, so I needed to duplicate it into the code base. It's not really a nice solution, but I think it would be an overkill to create a seperate fork / package just for that file. What do you think?

**Comment by brianlovin** [comment]
> What do you think?

Will let Max weigh in on this!

**Comment by mxstbr** [comment]
Rather than forking it submit the change as a PR to `draft-js-export-markdown`? https://github.com/sstur/draft-js-utils @sstur is very quick to merge PRs.

Also, that fix really has nothing to do with this PR, it should be done regardless!

Let's submit the patch as an upstream PR, revert the commit here and upgrade the dep in a spearate PR once the change is merged and published.

**Comment by Ponjimon** [comment]
So, I should do a PR at that project?

**Comment by mxstbr** [comment]
Yes, that one is perfect!

e2e test failure is unrelated, something to do with the new composer (cc @brianlovin), so I am going to ship this. Thanks so much for the great contribution @lookapanda!

**Pull request merged**
